### DISCUSSION TITLE
Better "Choose file" dialog

### DISCUSSION THREAD

**Discussion by ikyquocked** [linuxmint/discussions#538] [Ideas]
Hello! I have few "ideas" concerning a choosing file dialog. It's not something new actually, windows and macos have this since 2007 I guess...
![Screenshot from 2024-07-23 02-56-51](https://github.com/user-attachments/assets/c827ef11-0d4d-494d-9522-483ca235b0e9)

1. Searchbox. Damn god man, it's so easy to a find a file just using the filename and no mess with choosing a path
2. Pathbox. Users could paste the path to destination folder and not mess with clicking...

thats all, very easy things
How do you feel that?

**Comment by Jopp-gh** [comment]
1. There's no need for a search box: just start typing something, to search... this is valid for all Nemo windows 
2. The path bar is pretty useful because interactive, it isn't just a static info bar. Personally, I like the path bar.

After a few months of daily use you'll start to appreciate how Linux Mint is designed

**Reply by cplittlfield** [reply]
On the LM forums, there are lots of complaints about the GTK file chooser, and some of its features aren't obvious to new users.  It took me a year to discover the search capability, for example.  (It did occur to me to try Ctrl + H sooner than that, though. :smile: )

I do tend to agree, however, that being able to type a path is nice.  It's obvious that the OP hasn't tried Ctrl + L in the file chooser. :smile:

@ikyquocked  Try using Ctrl + L in the file chooser, the next time you want to change directories, and see if that gives you what you are looking for.  It probably won't be entirely satisfactory to use, since that entire dialogue box needs to be reworked, but you might find it enough to be getting on with.

On the other hand, if you need an OS that looks and behaves exactly like Windows, then you probably need to be using Windows.  Just saying.

**Reply by ikyquocked** [reply]
@cplittlfield @Jopp-gh 
ctrl+L works, but ctrl+H doesnt, but following this logic I tried ctrl+F and it works but only in current directory, not the entire drive!

But your suggestion about Windows is ridiculous, sorry.

Choosing file is a very basic feauture, everybody uses it billion times per day, and a lack of easy feautures are drilling brain. If I need a file but I dont know where is it it becomes a quest to find it, remember the path (sometimes it's long) and chose it in software...

**Reply by Jopp-gh** [reply]
I fully agree, digging a file tree to find files is tedious and antiquated, so I made a proposal time ago:
https://github.com/orgs/linuxmint/discussions/430

>But your suggestion about Windows is ridiculous, sorry

OK, this is opinion and we both tried to help you out. We aren't part of the Mint developer team. And even if we were, I'd suggest you to take it easier and say it differently. 

Switching to another Os isn't that smooth, you've to invest some time to re-learn some features. Different mindsets, different products.

**Reply by cplittlfield** [reply]
_"But your suggestion about Windows is ridiculous, sorry."_

Sorry, my comment was not a reflection on you personally.  Switching to a different OS and expecting the exact same behavior as what you are accustomed to is simply not reasonable; you have to be prepared for differences.  Yes, the same basic computing considerations obtain, but the DOS/Windows world and the UNIX/Linux world approach them from radically different viewpoints, so they have evolved different ways of handling things.

Not only that, but free and open-sourced software has a very different development model from that of proprietary software.  FOSS is not consumer driven; the devs write the sort of software _they_ want to use, leaving it up to the rest of us to take the source code and modify it to suit ourselves.  If that's not something we can or want to do, then we have to accept what the devs want to write.  They may be willing to take requests, but they are under no obligation to do so.

This all may feel unfair, but that's just how life works, and we all have to learn to cope.

**Comment by Qronikarz** [comment]
> Hello! I have few "ideas" concerning a choosing file dialog. It's not something new actually, windows and macos have this since 2007 I guess...

In Linux online world file chooser is a well known meme - https://wiki.installgentoo.com/wiki/File_Picker_meme
It's worth noting though that it finally got fixed/added in some GNOME version. But it needed GTK4.

So now Cinnamon and other GTK3 desktops continue this legacy. Here's a nemo feature request that would address all your points.
https://github.com/linuxmint/nemo/issues/3068
